Runner/Sandbox.cs

```diff
@@ -184,26 +167,19 @@ private void SetAppConfigIfExists()
 
         private void EnumerateAndLoadAssemblies()
         {
-            ScannedAssemblies = Directory.EnumerateFiles(Utils.GetGaugeBinDir(), "*.dll", SearchOption.TopDirectoryOnly)
-                .Select(s =>
-                {
-                    Logger.Debug("Loading assembly from : {0}", s);
-                    return Assembly.Load(AssemblyName.GetAssemblyName(s));
-                })
-                .ToList();
+            foreach (var path in Directory.EnumerateFiles(Utils.GetGaugeBinDir(), "*.dll", SearchOption.TopDirectoryOnly))
+                AssemblyScanner.TryAdd(path);
         }
 
         private void LocateTargetLibAssembly()
         {
-            TargetLibAssembly = ScannedAssemblies.First(assembly => assembly.GetName().Name == GaugeLibAssembleName);
+            TargetLibAssembly = typeof(Step).Assembly;
```

Comment:
This may cause a problem if different versions of `Lib` is loaded in Gauge.CSharp.Runner and the target test project. I.e. if Gauge.CSharp.Runner references Lib v0.5.2, whereas the test uses Lib v0.5.3, the two `Step` types are different. So reflection scanning would fail. Hence the original change to look for `TargetLibAssembly` using the `Assembly.Name`.